File: rookieone/node/src/day1/part1_spec.js
import assert from 'assert';

import { measureDepthIncreases } from './part1';
import { readPuzzleInput } from './readPuzzleInput';

describe("Day 1", () => {
  describe("Part 1", () => {
    it("example should return answer", () => {
      const input = readPuzzleInput('./src/day1/example.txt')

      const answer = measureDepthIncreases(input)

      assert.equal(answer, 7)
    })

    it("input should return answer", () => {
      const input = readPuzzleInput('./src/day1/input.txt')

      const answer = measureDepthIncreases(input)

      assert.equal(answer, 1791)
    })
  })
})

File: rookieone/ruby/spec/day4/part1_spec.rb
require "./code/day4/part1"
require "./code/day4/file_reader"

RSpec.describe "Day 4 Part 1" do
  it "example should return answer" do
    input = Day4FileReader.read_file("./spec/day4/example.txt")
    
    answer = Day4Part1.solve(input)

    expect(answer).to   eq(4512)
  end

  it "input should return answer" do
    input = Day4FileReader.read_file("./spec/day4/input.txt")
    
    answer = Day4Part1.solve(input)

    expect(answer).to   eq(11536)
  end
end
